mcp/mastra/verticals/shopping/triggers.ts

```diff
@@ -0,0 +1,180 @@
+
+Look for patterns like:
+- Items that were replaced with alternatives
+- Items that were removed from the order
+- Quantity changes
+- Price changes
+- Delivery time changes
+
+Always respond in Danish with a clear, concise summary.`,
+    description: 'Specialized agent for extracting order changes from Bilka notification emails',
+    tools: undefined,
+  },
+  inputSchema: emailInputSchema,
+  outputSchema: orderChangesSchema,
+  prompt: ({ inputData }) => {
+    // Email content is passed to the AI agent for extraction
+    // The structured output schema constrains what the agent can return
+    return `Analyser denne e-mail fra Bilka kundeservice og udtræk ændringerne i ordren:
+
+E-mail emne: ${inputData.email.subject}
+
+E-mail indhold:
+${inputData.email.bodyPreview}
+
+Besvar med:
+1. hasChanges: true hvis der er ændringer, false hvis ikke
+2. changes: En liste af specifikke ændringer (på dansk)
+3. summary: En kort opsummering af ændringerne (på dansk)
+
+Hvis du ikke kan identificere specifikke ændringer, så beskriv generelt hvad e-mailen handler om.`;
+  },
+});
+
+/**
+ * Sanitizes a string for use in voice notifications.
+ * Removes potentially harmful or unwanted characters/patterns.
+ */
+function sanitizeForVoiceNotification(text: string): string {
+  let sanitized = text;
+
+  // Remove HTML tags iteratively to handle nested/malformed tags
+  let previousLength: number;
+  do {
+    previousLength = sanitized.length;
+    sanitized = sanitized.replace(/<[^>]*>/g, '');
+  } while (sanitized.length < previousLength);
+
+  // Remove any remaining angle brackets to prevent partial tags
+  sanitized = sanitized.replace(/[<>]/g, '');
+
+  return (
+    sanitized
+      // Remove URLs
+      .replace(/https?:\/\/[^\s]+/g, '')
+      // Remove excessive whitespace
+      .replace(/\s+/g, ' ')
+      // Truncate to reasonable length for voice
+      .substring(0, 500)
+      .trim()
+  );
+}
+
+// Step 2: Send notification about order changes
+const notifyOrderChanges = createStep({
+  id: 'notify-order-changes',
+  description: 'Send notification about order changes via Home Assistant',
+  inputSchema: orderChangesSchema,
+  outputSchema: z.object({
+    notificationSent: z.boolean(),
+    message: z.string(),
+  }),
+  execute: async ({ inputData }) => {
+    if (!inputData.hasChanges) {
+      return {
+        notificationSent: false,
+        message: 'No order changes detected, skipping notification',
+      };
+    }
+
+    // Sanitize the AI-extracted content before using in voice notification
+    const sanitizedChanges = inputData.changes.map(sanitizeForVoiceNotification);
+    const sanitizedSummary = sanitizeForVoiceNotification(inputData.summary);
+
+    const notificationMessage =
+      sanitizedChanges.length > 0
+        ? `Sir, din Bilka ordre har følgende ændringer: ${sanitizedChanges.join('. ')}`
+        : `Sir, ${sanitizedSummary}`;
+
+    const result = await notifyDevice.execute({
```

Comment:
It shouldn't be notifyDevice that gets called. Instead, the whole notification agent should handle this. Therefore, the stuff provided to it should also just contain the email subject and body with its HTML removed and converted into plain text in a good way (not using regex, and maybe using a third party library).